src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -1953,6 +1974,14 @@ await UpdateUsersAsync(group, groupsDict[group.ExternalId].ExternalUserIds,
             }
         }
 
+        if (eventSystemUser != null)
+        {
+            await _eventService.LogOrganizationUserEventsAsync(events.Select(e => (e.ou, e.e, eventSystemUser.Value, e.d)));
+        }
+        else
+        {
+            await _eventService.LogOrganizationUserEventsAsync(events);
+        }
```

Comment:
Why does this need the `if` statement but `LogGroupEvents` doesn't?

Reply:
`LogGroupEvents` doesn't have an overload without an `EventSystemUser` argument, it just handles a null value for one.

```c#
    Task LogGroupEventsAsync(IEnumerable<(Group group, EventType type, EventSystemUser? systemUser, DateTime? date)> events);
```

`LogOragnizationUserEvents` uses an overload to exclude the property instead
```cs
    Task LogOrganizationUserEventsAsync<T>(IEnumerable<(T, EventType, DateTime?)> events) where T : IOrganizationUser;
    Task LogOrganizationUserEventsAsync<T>(IEnumerable<(T, EventType, EventSystemUser, DateTime?)> events) where T : IOrganizationUser;
```